web_widget_color/static/src/js/widget.js

```diff
@@ -1,53 +1,49 @@
     "use strict";
 
     var basic_fields = require('web.basic_fields');
     var field_registry = require('web.field_registry');
-    var FormController = require('web.FormController');
-    var field_utils = require('web.field_utils');
-
-    var _super_getDir = jscolor.getDir.prototype;
-    jscolor.getDir = function () {
-        var dir = _super_getDir.constructor();
-        if (dir.indexOf('web_widget_color') === -1) {
-            jscolor.dir = '/web_widget_color/static/lib/jscolor/';
-        }
-        return jscolor.dir;
-    };
+    var ListRenderer = require('web.ListRenderer');
 
     var FieldColor = basic_fields.FieldChar.extend({
         template: 'FieldColor',
         widget_class: 'oe_form_field_color',
-        _getValue: function() {
-            return field_utils.format.char(this.$('input').val());
+
+        _renderReadonly: function () {
+            // Do Nothing
         },
-        _render: function () {
-            var show_value = field_utils.format.char(this.value);
-            jscolor.init(this.$el[0]);
-            if (this.mode !== 'readonly') {
-                var $input = this.$el.find('input');
-                $input.val(show_value);
-                $input.css("background-color", show_value);
-                this.$input = $input;
-                this.$(".oe_form_char_content").text(show_value);
-                this.$('span').css("background-color", show_value);
-            } else {
-                this.$(".oe_form_char_content").text(show_value);
-                this.$('span').css("background-color", show_value);
 
-            }
+        _renderEdit: function() {
+            var self = this;
+            // FIXME: For some reason needs 1ms delay for initilize jscolor...
+            _.delay(function(){
+                self.$input = self.$el.find('input');
+                jscolor.installByClassName('jscolor');
+            }, 1);
         }
     });
     field_registry.add('color', FieldColor);
 
-    /*
-     * Init jscolor for each editable mode on view form
-     */
-    FormController.include({
-        _updateEnv : function (parentID) {
-            this._super(parentID);
-            jscolor.init(this.$el[0]);
-        }
+    // Deny unselect row when a field color has the focus
+    ListRenderer.include({
+        unselectRow: function () {
+            var canUnselect = true;
+            if (this.currentRow !== null) {
+                var record = this.state.data[this.currentRow];
+                var recordWidgets = this.allFieldWidgets[record.id];
+                _.each(recordWidgets, function (widget) {
+                    var $el = widget.getFocusableElement();
+                    if ($el.hasClass('jscolor') && $el.is(":focus")) {
+                        canUnselect = false;
+                        return false;
+                    }
+                });
+            }
+
+            return canUnselect?this._super.apply(this, arguments):$.when();
```

Comment:
maybe a normal IF/ELSE block would make this more readable?

Reply:
Can you instead of $.when() return a $.Deferred().resolve()?